ibis/pandas/client.py

```diff
@@ -354,41 +358,62 @@ def table(self, name, schema=None):
         return PandasTable(name, schema, self).to_expr()
 
     def execute(self, query, params=None, limit='default', **kwargs):
-        from ibis.pandas.core import execute_and_reset
 
         if limit != 'default':
             raise ValueError(
                 'limit parameter to execute is not yet implemented in the '
                 'pandas backend'
             )
 
-        assert isinstance(query, ir.Expr)
-        return execute_and_reset(query, params=params, **kwargs)
+        if not isinstance(query, ir.Expr):
+            raise TypeError(
+                "`query` has type {!r}, expected ibis.expr.types.Expr".format(
+                    type(query).__name__
+                )
+            )
+        return execute_last(
```

Comment:
why wouldn't this be part of the main `execute`? each client needs to call `execute_last`, and if i call the `execution` function rather than this method, i won't get the `execute_last` behavior.

Reply:
Because execute_last will always reset the index because of the default implementation. Window function execution requires the index to be preserved, and it makes recursive calls to `execute` which means we cannot reset the index.